relay-general/src/store/normalize/breakdowns.rs

```diff
@@ -105,11 +138,15 @@ impl EmitBreakdowns for SpanOperationsConfig {
             return None;
         }
 
-        let spans = match event.spans.value() {
-            None => return None,
-            Some(spans) => spans,
+        let mut spans = match event.spans.value() {
```

Comment:
instead of allocating a new vector here, can you extract the loop body into a function, and call it for each entry in `event.spans` + the root span?